tests/__init__.py

```diff
@@ -359,3 +362,66 @@ def assert_url_equal(url1, url2):
     assert_equal(parts1.hostname, parts2.hostname)
     assert_equal(parts1.port, parts2.port)
     assert_equal(parse_qs(parts1.query), parse_qs(parts2.query))
+
+
+class RawResponse(BytesIO):
+    def stream(self, **kwargs):
+        contents = self.read()
+        while contents:
+            yield contents
+            contents = self.read()
+
+
+class BotocoreHTTPStubber(object):
```

Comment:
Hmm maybe we should call it `ClientHTTPStubber` then? I think `Botocore` is still a little ambiguous because you can still make HTTP requests from the botocore layer via other mechanisms like credential resolving with IMDS and container metadata

Reply:
Yeah, I think that makes sense as this method does not stub those.